mssql/functions.py

```diff
@@ -343,7 +343,10 @@ def bulk_update_with_default(self, objs, fields, batch_size=None, default=0):
                     attr = Value(attr, output_field=field)
                 when_statements.append(When(pk=obj.pk, then=attr))
             if connection.vendor == 'microsoft' and value_none_counter == len(when_statements):
-                case_statement = Case(*when_statements, output_field=field, default=Value(default))
+                if default is None:
+                    case_statement = Case(*when_statements, output_field=field, default=F(field.attname))
+                else:
```

Comment:
I have been thinking and I don't fully get why it would be a good idea to keep this default value. If I understand properly, that value is never applied and only exists because SQL server syntax requires it. If that is correct, we are adding a branching that is not necessary.

Reply:
I agree that the intended usage of bulk update is that every associated obj has a corresponding value. However, since we introduced this field there could be users that rely on the else case as a default (as a result the default value can actually be applied). 

In our next major release we can notify users and depreciate the default functionality.

Reply:
@dauinsight I just realized that if all the cases set the value to null and we limit with our where that we only update the rows we have to update, we can replace the Case with Value(Null).
I mean,
`
UPDATE TABLE set col1 = CASE WHEN id = 1 THEN NULL when id = 2 THEN NULL DEFAULT col1 end
WHERE ID in (1,2)
`
can be simplified to
`
UPDATE TABLE set col1 = NULL
WHERE ID in (1,2)
`
Which is cleaner than what we are doing at the moment. What do you think?

Reply:
That's a very good point. We can omit the case statement entirely if we intend to set everything to NULL. 

I'll go ahead and make the change. Thanks!